modules/karpenter/variables.tf

```diff
@@ -107,7 +107,7 @@ variable "karpenter_nodeclasses" {
     karpenter_node_tags_map                = map(string)
     karpenter_ami_family                   = string
     karpenter_node_user_data               = string
-    karpenter_node_metadata_options        = map(string)
+    karpenter_node_metadata_options        = map(any)
```

Comment:
karpenter_node_metadata_options        = map(object({
        http_endpoint          = optional(string)
        http_tokens          = optional(string)
        http_put_response_hop_limit            = optional(int)
        instance_metadata_tags = optional(string)
        http_protocol_ipv6 = = optional(string)
})
)

Reply:
`karpenter_node_metadata_options        = map(object({
        httpEndpoint          = optional(string)
        httpTokens          = optional(string)
        httpPutResponseHopLimit            = optional(int)
        instanceMetadataTags = optional(string)
        httpProtocolIpv6 = = optional(string)
})
)  I think so this will work, let me try

Reply:
Int is not supported type, i need to try with optional(number)